lib/teiserver/data/cache_user.ex

```diff
@@ -1007,6 +1009,56 @@ defmodule Teiserver.CacheUser do
     end
   end
 
+  @spec tachyon_login(T.user(), String.t(), String.t()) ::
```

Comment:
This is mostly taken from [login_from_token](https://github.com/beyond-all-reason/teiserver/pull/378/files#diff-0cb27f1af2cc6c1a7428ea44636b9db9c9fbc313f7889d75a9f2fd3d33b18accL826-L906)